hack/jenkins/e2e.sh

```diff
@@ -225,9 +225,16 @@ fi
 # TODO(zml): We have a bunch of legacy Jenkins configs that are
 # expecting junit*.xml to be in ${WORKSPACE} root and it's Friday
 # afternoon, so just put the junit report where it's expected.
-for junit in ${ARTIFACTS}/junit*.xml; do
-  ln -s ${junit} ${WORKSPACE}
-done
+results=($(ls "${ARTIFACTS}" | grep "junit.*.xml")) || true
+if [[ ! -z ${results:-} ]]; then
+    for junit in "${results[@]}"; do
+        # Jenkins runs that trigger sequentially (e.g. step1 -> step2 -> ...) will
+        # have e2e test results overwrite each other because they all run in the
+        # same workspace. For those cases, we prefix the final location with the
+        # step name.
+        mv "${ARTIFACTS}/${junit}" "${WORKSPACE}/${JENKINS_STEP_PREFIX:-}${junit}"
```

Comment:
If we move the junit files out of the artifacts directory, then the GCS uploader won't find them anymore.
A better fix would be either to set a separate artifacts directory for each stage, or to update test/e2e/e2e_test.go to allow specifying a slightly different junit filename.

As Zach's comment above notes, we should probably eventually stop symlinking the junit files into the root workspace directory entirely.